src/OpenTelemetry.Instrumentation.Http/Implementation/HttpWebRequestActivitySource.netfx.cs

```diff
@@ -265,36 +270,25 @@ private static bool IsRequestInstrumented(HttpWebRequest request)
 
     private static void ProcessRequest(HttpWebRequest request)
     {
-        if (!WebRequestActivitySource.HasListeners() || !Options.EventFilterHttpWebRequest(request))
-        {
-            // No subscribers to the ActivitySource or User provider Filter is
-            // filtering this request.
-            // Propagation must still be done in such cases, to allow
-            // downstream services to continue from parent context, if any.
-            // Eg: Parent could be the Asp.Net activity.
-            InstrumentRequest(request, Activity.Current?.Context ?? default);
-            return;
-        }
-
         if (IsRequestInstrumented(request))
         {
             // This request was instrumented by previous
             // ProcessRequest, such is the case with redirect responses where the same request is sent again.
             return;
         }
 
-        var activity = WebRequestActivitySource.StartActivity(ActivityName, ActivityKind.Client);
+        // No subscribers to the ActivitySource or User provider Filter is
+        // filtering this request.
+        var skipTracing = !WebRequestActivitySource.HasListeners() || !Options.EventFilterHttpWebRequest(request);
+        var activity = skipTracing
+            ? null
+            : WebRequestActivitySource.StartActivity(ActivityName, ActivityKind.Client);
```

Comment:
```suggestion
        Activity activity = null;
        if (!WebRequestActivitySource.HasListeners() || !Options.EventFilterHttpWebRequest(request))
        {
            activity = WebRequestActivitySource.StartActivity(ActivityName, ActivityKind.Client);
        }
```
One more line, but maybe a bit easier to read.